rust/processor/src/models/token_v2_models/v2_token_ownerships.rs

```diff
@@ -294,69 +294,78 @@ impl TokenOwnershipV2 {
-                            tracing::error!(
-                                transaction_version = txn_version,
-                                lookup_key = &token_address,
-                                "Failed to find NFT for burned token. You probably should backfill db."
-                            );
-                            return Ok(None);
-                        },
-                    }
-                },
+            // 1. Try to lookup token address in burn event mapping
+            let previous_owner = if let Some(burn_event) =
+                tokens_burned_map.get(&standardize_address(&write_resource.address.to_string()))
+            {
+                burn_event.get_previous_owner_address()
+            } else {
+                // 2. If it doesn't exist in burn event mapping, then it must be an old burn event that doesn't contain previous_owner.
+                // Do a lookup to get preivous owner.
+                let latest_nft_ownership = match prior_nft_ownership.get(token_address) {
+                    Some(inner) => inner.clone(),
+                    None => {
+                        match CurrentTokenOwnershipV2Query::get_latest_owned_nft_by_token_data_id(
+                            conn,
+                            token_address,
+                        )
+                        .await
+                        {
+                            Ok(nft) => nft,
+                            Err(_) => {
+                                tracing::error!(
+                                    transaction_version = txn_version,
+                                    lookup_key = &token_address,
+                                    "Failed to find NFT for burned token. You probably should backfill db."
+                                );
+                                return Ok(None);
+                            },
+                        }
+                    },
+                };
+                latest_nft_ownership.owner_address.clone()
             };
 
             let token_data_id = token_address.clone();
-            let owner_address = latest_nft_ownership.owner_address.clone();
             let storage_id = token_data_id.clone();
-            let is_soulbound = latest_nft_ownership.is_soulbound;
 
             return Ok(Some((
                 Self {
                     transaction_version: txn_version,
                     write_set_change_index,
                     token_data_id: token_data_id.clone(),
                     property_version_v1: BigDecimal::zero(),
-                    owner_address: Some(owner_address.clone()),
+                    owner_address: Some(previous_owner.clone()),
                     storage_id: storage_id.clone(),
                     amount: BigDecimal::zero(),
                     table_type_v1: None,
                     token_properties_mutated_v1: None,
-                    is_soulbound_v2: is_soulbound,
+                    is_soulbound_v2: Some(false),
```

Comment:
@bowenyang007 is this ok? or should we make this `None`?

Reply:
Once we have better deletion handling, I guess this doesn't matter much

Reply:
actually should we just add deletion handling here as well?

Reply:
i'll give it a shot

Reply:
actually let's leave deletion handling for later. there's some edge cases we need to think about, like if the row doesn't exist yet.

Reply:
this data is wrong though.

Reply:
The simplest thing to do is to have another field called burnt_transaction_version and update that field when the token is burnt. That way the existing queries would work as intended because you're not updating last_transaction-version

Reply:
so the idea is to have to the deletion query handle the `amount` field and the insertion query to handle updating the rest of the fields. how do we set `amount = 1` the first time and not the other times?

Reply:
we could split the insertion query into 2 queries but that would double the # of insertion queries

Reply:
a race condition between the mint/transfer and burn insertions can happen and cause some fields to be wrong, but we sync'd offline that data accuracy for burnt tokens/objects is best effort so this is fine. gonna consider this resolved.